fairlearn/utils/_input_validation.py

```diff
@@ -125,12 +125,7 @@ def _validate_and_reformat_input(
     else:
         result_y = pd.Series(dtype="float64")
 
-    return (
-        pd.DataFrame(X),
-        result_y,
-        sensitive_features,
-        control_features,
-    )
+    return (X, result_y, sensitive_features, control_features)
```

Comment:
`X` can no longer be converted to a `DataFrame`, because it's no longer guaranteed to be tabular.

I double checked that the returned `X` of this function is never used in the code-base so this change is safe.